conda_libmamba_solver/solver.py

```diff
@@ -797,3 +881,12 @@ def _check_spec_compat(self, match_spec):
     def _reset(self):
         self.solver = None
         self._solver_options = None
+
+    def _called_from_conda_build(self):
+        """
+        conda build calls the solver via `conda.plan.install_actions`, which
+        overrides Solver._index (populated in the classic solver, but empty for us)
+        with a custom index. We can use this to detect whether conda build is in use
+        and apply some compatibility fixes.
+        """
+        return "conda_build" in sys.modules and getattr(self, "_index", None)
```

Comment:
This might make sense to use the stdlib [`inspect.currentframe()`](https://docs.python.org/3/library/inspect.html#inspect.currentframe) here instead of checking for `sys.modules` since the latter might be true for other weird reasons and could create a false positive.

E.g.:
```suggestion
        return "install_actions" in inspect.currentframe().f_back.f_code.co_name and getattr(self, "_index", None)
```